src/core/lib/transport/handshaker_factory.h

```diff
@@ -35,11 +37,38 @@ namespace grpc_core {
 
 class HandshakeManager;
 
+// Enum representing the priority of the client handshakers.
+// The order of the client handshakers is decided by the priority.
+// For example kPreTCPConnect handshakers are called before kTCPConnect and so
+// on.
+enum class HandshakerClientPriority : int {
+  kPreTCPConnect,
```

Comment:
Probably worth expanding the name and including one line on the kinds of things expected in each bucket

Reply:
Done, let me know if this good enough or if I should add more information.